src/main/java/edu/cornell/kfs/sys/batch/MoveFileStep.java

```diff
@@ -0,0 +1,128 @@
+package edu.cornell.kfs.sys.batch;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.sys.batch.AbstractStep;
+
+import edu.cornell.kfs.sys.CUKFSConstants;
+
+/**
+ * Generic batch step for moving files from one directory to another.
+ * 
+ * At a minimum, the sourcePath and targetPath must be set to valid pre-existing directories,
+ * and the fileNamePattern must be set to a valid Java regex String. Only files whose names
+ * match the regex will be moved from the source directory to the target directory.
+ * This step will also automatically create ".done" files in the target directory for each moved file.
+ * 
+ * The source/target directory paths are not required to have trailing slashes. This step
+ * will automatically add them as needed.
+ */
+public class MoveFileStep extends AbstractStep {
+
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(MoveFileStep.class);
+
+    protected String sourcePath;
+    protected String targetPath;
+    protected Pattern fileNamePattern;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        if (StringUtils.isBlank(sourcePath)) {
+            throw new IllegalStateException("sourcePath cannot be blank");
+        } else if (StringUtils.isBlank(targetPath)) {
+            throw new IllegalStateException("targetPath cannot be blank");
+        } else if (fileNamePattern == null) {
+            throw new IllegalStateException("fileNamePattern cannot be null");
+        }
+        
+        super.afterPropertiesSet();
+    }
+
+    @Override
+    public boolean execute(String jobName, Date jobRunDate) throws InterruptedException {
+        return moveFiles();
+    }
+
+    protected boolean moveFiles() {
+        verifyDirectoriesExist(sourcePath, targetPath);

```

Comment:
I tried using FileSystemFileStorageServiceImpl for implementing many of the different operations for this. However, the service assumes that either the directory path is an absolute one, or that the path should be prefixed with another absolute or relative path. I can look into using that service further if desired, but I would likely need to create a separately-configured service bean that has an appropriate path prefix configured.